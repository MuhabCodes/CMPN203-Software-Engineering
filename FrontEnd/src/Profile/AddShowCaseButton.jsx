import React from 'react';
import add from './assets/add-icon.png';
import './Showcase.css';
/**
 * This component renders the add to showcase button
 * @component AddShowCaseButton
 * @function AddShowCaseButton
 * @example <AddShowCaseButton/>
 * @returns button
 */
const AddShowCaseButton = () => (
  <button type="button" className="add-btn-scp" data-testid="addsc-btn">
    <img src={add} className="add-icon-scp" alt="" />
  </button>
);

export default AddShowCaseButton;
