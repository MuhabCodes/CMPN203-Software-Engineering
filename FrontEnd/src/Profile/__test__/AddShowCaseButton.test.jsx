import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import AddShowCaseButton from '../AddShowCaseButton';
import '@testing-library/jest-dom/extend-expect';

it('it renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<AddShowCaseButton />, div);
});

it('renders the button correctly', () => {
  const { getByTestId } = render(<AddShowCaseButton />);
  expect(getByTestId('addsc-btn')).toBeTruthy();
});

it('matches a snapshot', () => {
  const tree = renderer.create(<AddShowCaseButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
