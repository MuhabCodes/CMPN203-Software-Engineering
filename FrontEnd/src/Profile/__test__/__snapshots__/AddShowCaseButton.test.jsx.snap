// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a snapshot 1`] = `
<button
  className="add-btn-scp"
  data-testid="addsc-btn"
  type="button"
>
  <img
    alt=""
    className="add-icon-scp"
    src="add-icon.png"
  />
</button>
`;
